tap_s3_csv/transform.py

```diff
@@ -317,20 +318,18 @@ def _transform(self, data, typ, schema, path):
                 return False, None
 
         elif typ == "integer":
-            if isinstance(data, str):
-                data = data.replace(",", "")
-
             try:
-                return True, int(data)
+                num = data.replace(',', '') if isinstance(data, str) else data
+                int(num)    # check if it can be cast successfully
+                return True, str(data)
             except:
                 return False, None
 
         elif typ == "number":
-            if isinstance(data, str):
-                data = data.replace(",", "")
-
             try:
-                return True, float(data)
+                num = data.replace(',', '') if isinstance(data, str) else data
+                float(num)
+                return True, str(data)
```

Comment:
@ali-varicent , @kchan-varicent , @yi-varicent , @woody-feng ,
In the `tap-tqp` import (which involves `tap-s3-csv` and `target-s3-csv`) this causes a failure (it started failing after these changes had been merged). A JIRA ticket: https://varicent.atlassian.net/browse/WP-10970
@ali-varicent , could you follow up on this, please, while I'm away?